allennlp/service/predictors/wikitables_parser.py

```diff
@@ -36,3 +57,42 @@ def _json_to_instance(self, json_dict: JsonDict) -> Tuple[Instance, JsonDict]:
                                                          tokenized_question=tokenized_question)
         extra_info = {'question_tokens': tokenized_question}
         return instance, extra_info
+
+    @overrides
+    def predict_json(self, inputs: JsonDict, cuda_device: int = -1) -> JsonDict:
+        instance, return_dict = self._json_to_instance(inputs)
+        outputs = self._model.forward_on_instance(instance, cuda_device)
+
+        logical_form_filename = os.path.join(SEMPRE_DIR, 'logical_forms.txt')
+        with open(logical_form_filename, 'w') as f:
+            f.write(outputs['logical_form'] + '\n')
+
+        table_dir = SEMPRE_DIR + 'csv/'
+        os.makedirs(table_dir, exist_ok=True)
+        table_filename = 'context.csv'
+        with open(table_dir + table_filename, 'w', encoding='utf-8') as f:
+            f.write(inputs["table"])
+
+        test_record = ('(example (id %s) (utterance %s) (context (graph tables.TableKnowledgeGraph %s))'
+                       '(targetValue (list (description "6"))))' % ('nt-0', inputs['question'],
+                                                                     table_filename))
+        test_data_filename = SEMPRE_DIR + 'data.examples'
+        with open(test_data_filename, 'w') as f:
+            f.write(test_record)
+
+        command = ' '.join(['java',
+                            '-jar',
+                            cached_path(DEFAULT_EXECUTOR_JAR),
+                            test_data_filename,
+                            logical_form_filename,
+                            table_dir,
+                            ])
+        run(command, shell=True)
+
+        denotations_file = SEMPRE_DIR + 'logical_forms_denotations.tsv'
+        with open(denotations_file) as f:
+            line = f.readline().split('\t')
+            outputs['answer'] = line[1] if len(line) > 1 else line[0]
```

Comment:
Can you also modify the predictor test to make sure that we're getting an answer in the output?

Reply:
done.